elastic/kibana
# [APM][OTel] Use `telemetry.sdk` as a fallback for missing `agent.name` on non-tracing data

## Summary

Closes #195854

This PR adds a fallback when we are missing `agent.name` on APM Service Inventory list.
Using [OTel semantic convention fields](https://opentelemetry.io/docs/specs/semconv/resource/#telemetry-sdk) `telemetry.sdk.language` and `telemetry.sdk.name` to maintain `agent.name` format for OTel fields like `otlp/${agent}` `opentelemetry/${agent}`.

## Screenshots
| Before                                                                                    | After                                                                                     |
|-------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------|
| ![image](https://github.com/user-attachments/assets/792ca256-f01d-4eae-8a2d-af16fa34ea61) | ![image](https://github.com/user-attachments/assets/2816d1c7-1207-4da8-adb5-ec417b3fd26e) |
![image](https://github.com/user-attachments/assets/27df0ffc-8d5f-475c-ad6b-04086521871b)|![image](https://github.com/user-attachments/assets/d088a746-1375-4918-8e55-d3968a80772d)


## How to test it
1. Open otel-apm-e2e-poc repo
2. Add the following code under `processors -> transform -> metric_statements` in `otelcol.yaml`
```yaml
    - context: datapoint
      statements:
        - set(attributes["processor.event"], "metric")
```
3. Run `make start-stack` & `make build` & `make run` on otel poc repo 
4. Don't forget to run also `make start` on opentelemetry-demo and [follow this guide](https://github.com/elastic/otel-apm-e2e-poc?tab=readme-ov-file#working-with-opentelemetry-demo) to make it work.
5. In Kibana add `elasticsearch.ignoreVersionMismatch: true` to `kibana.dev.yml`.
6. If you go to APM Service Inventory list you will see missing icons.
7. Checkout to my branch
8. Try again, they should be fixed
